ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java

```diff
@@ -0,0 +1,218 @@
+                    .id(KONTAKTLINSER_VANSTER_OGA_DELSVAR_ID_8)
+                    .selected(
+                        vansterOga != null && vansterOga.getKontaktlins() != null ? vansterOga.getKontaktlins() : null)
+                    .build()
+            )
+            .build();
+    }
+
+    private static CertificateDataValueVisualAcuity getRightEyeValue(Synskarpevarden hogerOga) {
+        return CertificateDataValueVisualAcuity.builder()
+            .withoutCorrection(
+                CertificateDataValueDouble.builder()
+                    .id(HOGER_OGA_UTAN_KORREKTION_DELSVAR_ID_8)
+                    .number(
+                        hogerOga != null && hogerOga.getUtanKorrektion() != null ? hogerOga.getUtanKorrektion() : null)
+                    .build()
+            )
+            .withCorrection(
+                CertificateDataValueDouble.builder()
+                    .id(HOGER_OGA_MED_KORREKTION_DELSVAR_ID_8)
+                    .number(
+                        hogerOga != null && hogerOga.getMedKorrektion() != null ? hogerOga.getMedKorrektion() : null)
+                    .build()
+            )
+            .binocular(
+                CertificateDataValueBoolean.builder()
+                    .id(KONTAKTLINSER_HOGER_OGA_DELSVAR_ID_8)
+                    .selected(
+                        hogerOga != null && hogerOga.getKontaktlins() != null ? hogerOga.getKontaktlins() : null)
+                    .build()
+            )
+            .build();
+    }
+
+    private static VisualAcuity getBinocularConfig(CertificateTextProvider textProvider) {
+        return VisualAcuity.builder()
+            .label(textProvider.get(BINOKULART_LABEL_ID))
+            .withoutCorrectionId(BINOKULART_UTAN_KORREKTION_DELSVAR_ID_8)
+            .withCorrectionId(BINOKULART_MED_KORREKTION_DELSVAR_ID_8)
+            .build();
+    }
+
+    private static VisualAcuity getLeftEyeConfig(CertificateTextProvider textProvider) {
+        return VisualAcuity.builder()
+            .label(textProvider.get(VANSTER_OGA_LABEL_ID))
+            .withoutCorrectionId(VANSTER_OGA_UTAN_KORREKTION_DELSVAR_ID_8)
+            .withCorrectionId(VANSTER_OGA_MED_KORREKTION_DELSVAR_ID_8)
+            .contactLensesId(KONTAKTLINSER_VANSTER_OGA_DELSVAR_ID_8)
+            .build();
+    }
+
+    private static VisualAcuity getRightEyeConfig(CertificateTextProvider textProvider) {
+        return VisualAcuity.builder()
+            .label(textProvider.get(HOGER_OGA_LABEL_ID))
+            .withoutCorrectionId(HOGER_OGA_UTAN_KORREKTION_DELSVAR_ID_8)
+            .withCorrectionId(HOGER_OGA_MED_KORREKTION_DELSVAR_ID_8)
+            .contactLensesId(KONTAKTLINSER_HOGER_OGA_DELSVAR_ID_8)
+            .build();
+    }
+
+    public static Synskarpevarden toInternal(Certificate certificate, SynskarpaEnum type) {
+        final var value = (CertificateDataValueVisualAcuities) certificate.getData().get(VARDEN_FOR_SYNSKARPA_ID).getValue();
```

Comment:
Return empty Synskarpevarden if getData().get returns null

Reply:
Check!

Reply:
Sorry for not being completely clear in my comment... My comment was to add a null check on certificate.getData().get(VARDEN_FOR_SYNSKARPA_ID) 

But maybe it is best to make sure do check null for getValue() as well